src/backend/common/err_common.hpp

```diff
@@ -139,16 +139,14 @@ void print_error(const std::string& msg);
                       ERR_TYPE);                                           \
     } while (0)
 
-#define AF_RETURN_ERROR(MSG, ERR_TYPE)                                       \
-    do {                                                                     \
-        AfError err(__PRETTY_FUNCTION__, __AF_FILENAME__, __LINE__, MSG,     \
-                    ERR_TYPE);                                               \
-        std::stringstream s;                                                 \
-        s << "Error in " << err.getFunctionName() << "\n"                    \
-          << "In file " << err.getFileName() << ":" << err.getLine() << "\n" \
-          << err.what() << "\n";                                             \
-        print_error(s.str());                                                \
-        return ERR_TYPE;                                                     \
+#define AF_RETURN_ERROR(MSG, ERR_TYPE)                                \
+    do {                                                              \
+        std::stringstream s;                                          \
+        s << "Error in " << __PRETTY_FUNCTION__ << "\n"               \
+          << "In file " << __AF_FILENAME__ << ":" << __LINE__ << ": " \
+          << MSG;                                                     \
+        set_error(s.str(), ERR_TYPE);                                 \
+        return ERR_TYPE;                                              \
```

Comment:
Last two lines could be merged to avoid double editing in future.
`return set_error(s.str(), ERR_TYPE)`